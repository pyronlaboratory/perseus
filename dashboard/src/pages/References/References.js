import React from 'react'

// material imports
import { withStyles } from '@material-ui/core/styles';

import AppBar from '@material-ui/core/AppBar';
import CssBaseline from '@material-ui/core/CssBaseline';
import Divider  from '@material-ui/core/Divider';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';

const styles = (theme) => ({
  secondaryBar: {
    zIndex: 0,
    display: 'flex',
    height: '6.85em',
    justifyContent: 'center'
  },
  wrapper: {
    padding: theme.spacing(6, 4),
    [theme.breakpoints.down('md')]: {
        padding: theme.spacing(3, 1),
    },
  },
  paper: {
    maxWidth: 1064,
    margin: 'auto',
    overflow: 'hidden',
    padding: theme.spacing(3, 3),
    [theme.breakpoints.down('md')]: {
        padding: theme.spacing(2, 2),
    },
  },
  title: {
    margin: theme.spacing(3, 0),
    [theme.breakpoints.down('md')]: {
        margin: theme.spacing(2, 0),
    },
  },
  subtitle: {
    marginBottom: 10
  },
  text: {
    fontSize: 14
  },
  divider: {
    height: '0.5px',
    margin: '15px',
    backgroundColor: '#d4d4d4'
  }
});

// TO DO: Refactor: Move descriptive data to a json
function References(props) {
    const { classes } = props;

    return (
        <div>
            <AppBar
                component="div"
                className={classes.secondaryBar}
                style={{ backgroundColor: "#18202c"}}
                position="static"
                elevation={0}
            >
                <Toolbar>
                <Grid container alignItems="center" spacing={1}>
                    <Grid item xs>
                    <Typography color="inherit" variant="h5" component="h1">
                        References
                    </Typography>
                    </Grid>
                </Grid>
                </Toolbar>
            </AppBar>
            <div className={classes.wrapper} >
            <Paper className={classes.paper}>
                <Typography component="span">

                    <b>Following online resources were used during the development of this case study</b>
                    <Divider className={classes.divider} />
                    <CssBaseline />
                    <code>React.js</code>: 
                    <Typography className={classes.text} variant="body1" gutterBottom>
                        A JavaScript library for building user interfaces. <br />https://reactjs.org/ <br /><br />
                    </Typography>

                    <code>Material UI</code>: 
                    <Typography className={classes.text} variant="body1" gutterBottom>
                        React components for faster and easier web development. Build your own design system, or start with Material Design. <br />https://material-ui.com/guides/api/ <br /><br />
                    </Typography>
                    
                    <code>Flask</code>: 
                    <Typography className={classes.text} variant="body1" gutterBottom>
                        Python based microframework for web development. <br />https://flask.palletsprojects.com/en/2.0.x/<br /><br/>
                    </Typography>

                    <code>Heroku</code>: 
                    <Typography className={classes.text} variant="body1" gutterBottom>
                        Learn about building, deploying, and managing your apps on Heroku. Heroku makes it easy to deploy and scale Python apps. Whether you prefer frameworks like Django or Flask, or getting your hands dirty with Twisted or raw sockets, Heroku helps you build things your way with the tools you love.<br />https://devcenter.heroku.com/<br /><br />
                    </Typography>

                    <code>Firebase</code>: 
                    <Typography className={classes.text} variant="body1" gutterBottom>
                        Firebase helps you build and run successful apps. This application is running on Firebase.<br />https://firebase.google.com/docs<br /><br />
                    </Typography>

                    <code>Docker</code>: 
                    <Typography className={classes.text} variant="body1" gutterBottom>
                    Build safer, share wider, run faster.<br />https://docs.docker.com/<br /><br />
                    </Typography>
 
                    <Typography className={classes.text} variant="body1" gutterBottom>
                        and good old.. <br /><br />
                    </Typography>

                    <code>StackOverflow</code>: 
                    <Typography className={classes.text} variant="body1" gutterBottom>
                        https://stackoverflow.com/ <br /><br />
                    </Typography>

                </Typography>
                <CssBaseline />            
            </Paper>
            </div>
        </div>
    )
}

export default withStyles(styles)(References)
