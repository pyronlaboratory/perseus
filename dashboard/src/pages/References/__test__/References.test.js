import React from 'react';
import ReactDOM from 'react-dom';

// react testing
import { cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import References from '../References';

afterEach(cleanup);

describe('references page', () => {
    it("renders without crashing", () => {
        const div = document.createElement("div");
        ReactDOM.render(<References />, div);
    })

    it("matches snapshot", () => {
        const tree = renderer.create(<References />).toJSON();
        expect(tree).toMatchSnapshot();
    })
})
