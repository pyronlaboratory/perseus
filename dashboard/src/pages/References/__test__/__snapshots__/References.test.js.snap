// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`references page matches snapshot 1`] = `
<div>
  <div
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary References-secondaryBar-1 MuiPaper-elevation0"
    style={
      Object {
        "backgroundColor": "#18202c",
      }
    }
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <div
        className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 MuiGrid-align-items-xs-center"
      >
        <div
          className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true"
        >
          <h1
            className="MuiTypography-root MuiTypography-h5 MuiTypography-colorInherit"
          >
            References
          </h1>
        </div>
      </div>
    </div>
  </div>
  <div
    className="References-wrapper-2"
  >
    <div
      className="MuiPaper-root References-paper-3 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <span
        className="MuiTypography-root MuiTypography-body1"
      >
        <b>
          Following online resources were used during the development of this case study
        </b>
        <hr
          className="MuiDivider-root References-divider-7"
        />
        <code>
          React.js
        </code>
        :
        <p
          className="MuiTypography-root References-text-6 MuiTypography-body1 MuiTypography-gutterBottom"
        >
          A JavaScript library for building user interfaces. 
          <br />
          https://reactjs.org/ 
          <br />
          <br />
        </p>
        <code>
          Material UI
        </code>
        :
        <p
          className="MuiTypography-root References-text-6 MuiTypography-body1 MuiTypography-gutterBottom"
        >
          React components for faster and easier web development. Build your own design system, or start with Material Design. 
          <br />
          https://material-ui.com/guides/api/ 
          <br />
          <br />
        </p>
        <code>
          Flask
        </code>
        :
        <p
          className="MuiTypography-root References-text-6 MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Python based microframework for web development. 
          <br />
          https://flask.palletsprojects.com/en/2.0.x/
          <br />
          <br />
        </p>
        <code>
          Heroku
        </code>
        :
        <p
          className="MuiTypography-root References-text-6 MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Learn about building, deploying, and managing your apps on Heroku. Heroku makes it easy to deploy and scale Python apps. Whether you prefer frameworks like Django or Flask, or getting your hands dirty with Twisted or raw sockets, Heroku helps you build things your way with the tools you love.
          <br />
          https://devcenter.heroku.com/
          <br />
          <br />
        </p>
        <code>
          Firebase
        </code>
        :
        <p
          className="MuiTypography-root References-text-6 MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Firebase helps you build and run successful apps. This application is running on Firebase.
          <br />
          https://firebase.google.com/docs
          <br />
          <br />
        </p>
        <code>
          Docker
        </code>
        :
        <p
          className="MuiTypography-root References-text-6 MuiTypography-body1 MuiTypography-gutterBottom"
        >
          Build safer, share wider, run faster.
          <br />
          https://docs.docker.com/
          <br />
          <br />
        </p>
        <p
          className="MuiTypography-root References-text-6 MuiTypography-body1 MuiTypography-gutterBottom"
        >
          and good old.. 
          <br />
          <br />
        </p>
        <code>
          StackOverflow
        </code>
        :
        <p
          className="MuiTypography-root References-text-6 MuiTypography-body1 MuiTypography-gutterBottom"
        >
          https://stackoverflow.com/ 
          <br />
          <br />
        </p>
      </span>
    </div>
  </div>
</div>
`;
